spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2285,15 +2261,31 @@ object QueryPlanSerde extends Logging with CometExprShim {
         }
 
       case op =>
-        // Emit warning if:
-        //  1. it is not Spark shuffle operator, which is handled separately
-        //  2. it is not a Comet operator
-        if (!op.nodeName.contains("Comet") && !op.isInstanceOf[ShuffleExchangeExec]) {
-          val msg = s"unsupported Spark operator: ${op.nodeName}"
-          emitWarning(msg)
-          withInfo(op, msg)
+        opSerdeMap.get(op.getClass) match {
+          case Some(handler) =>
+            handler.enabledConfig match {
+              case Some(enabledConfig) =>
+                if (!enabledConfig.get(op.conf)) {
+                  withInfo(
+                    op,
+                    s"Native support for operator ${op.getClass.getSimpleName} is disabled. " +
+                      s"Set ${enabledConfig.key}=true to enable it.")
+                  return None
+                }
+              case _ =>
+            }
```

Comment:
this probably can be simplified

```suggestion
opSerdeMap.get(op.getClass).foreach { handler =>
  handler.enabledConfig.foreach { cfg =>
    if (!cfg.get(op.conf)) {
      withInfo(
        op,
        s"Native support for operator ${op.getClass.getSimpleName} is disabled. " +
          s"Set ${cfg.key}=true to enable it."
      )
      return None
    }
  }
}
```